relayer/pkg/chainlink/ocr2/config_digester.go

```diff
@@ -59,15 +59,15 @@ func (d offchainConfigDigester) ConfigDigest(cfg types.ContractConfig) (types.Co
 
 	offchainConfig := EncodeBytes(cfg.OffchainConfig)
 
-	// TODO: use libocr's custom encoding instead to produce encodedOnchainConfig
-	onchainConfig, err := median.DecodeOnchainConfig(cfg.OnchainConfig)
+	onchainConfig, err := medianreport.OnchainConfigCodec{}.DecodeBigInt(cfg.OnchainConfig)
 	if err != nil {
 		return configDigest, err
 	}
-	encodedOnchainConfig := []*big.Int{
-		big.NewInt(1), // onchainConfigVersion
-		new(big.Int).Mod(onchainConfig.Min, caigotypes.MaxFelt.Big()), // TODO: this wraps negative values correctly into felts, use a helper
-		new(big.Int).Mod(onchainConfig.Max, caigotypes.MaxFelt.Big()), // TODO: this wraps negative values correctly into felts, use a helper
+
+	encodedOnchainConfig := []*big.Int{}
+	for i := 0; i < len(onchainConfig); i++ {
+		modded := new(big.Int).Mod(onchainConfig[i], caigotypes.MaxFelt.Big()) // TODO: this wraps negative values correctly into felts, use a helper
```

Comment:
This shouldn't be done here, and not for all values. Only answers, observations & min/max answer are signed values. Make this a part of the encode operation.

Reply:
@krebernisak you are correct, the main thing is that onchainconfig is hashed as individual components because of using felts - so `encodedOnchainConfig` in this sense is improperly named since it's actually the OnchainConfig decoded into `*big.int` components